### ISSUE TITLE
Figure out why Quick Edit refuses to disappear

### ISSUE THREAD

**Issue by KeniMorri** [humphd/brackets#125]
Quick edit = Cmd + E

Presently if we expose a Tag, ID, or Class with quick edit that presently does no actually link to any local css files, when we try to close it with cmd+e it goes away for a brief second, then reappears immediately.

The colour picker is unaffected by this, it works as expected.

Extra notes: You can remove quick edit with esc, or by clicking the close button though.

**Comment by AJDBenner** [comment]
From my testing, quick edit only comes back immediately after closing it if there isn't any style included on the page.

**Comment by KeniMorri** [comment]
Alright with a little bit of testing I've found the general issue with why quick edit refuses to disappear.
So to start I'll show off what happens in brackets presently:

![brackets](https://cloud.githubusercontent.com/assets/1750230/6805068/7930476c-d217-11e4-8f66-61daf2b77f7b.gif)

In this gif you can see me Expose the p tag, then encircle where the cursor presently is,
Then I expose the ol tag, then encircle where the cursor presently is.

As opposed to bramble presently, where I do the same steps:
![bramble](https://cloud.githubusercontent.com/assets/1750230/6805102/a345cc66-d217-11e4-938a-1f0f289fc9dd.gif)

As you can see from the gif, "new rule" is not our current focus, our current focus when I expose it is in code mirror, thus when I expose again, I'm exposing from within code mirror again, thus why it disappears, and reappears.

Cmd + e will make the quick edit disappear, but if cmd + e is pressed within an element we have already exposed then it'll disappear then reappear.

TL:DR: When exposing element with no CSS applied focus is not put into the quick edit panel, it remains on the element, thus if we press it again it'll disappear then reappear.

**Comment by AJDBenner** [comment]
When you disable the HTMLHinter extension this issue still persists.

**Comment by AJDBenner** [comment]
From my research this is what is happening.

When you hit CMD + E [this](https://github.com/humphd/brackets/blob/bramble/src/editor/EditorManager.js#L294) function is called which calls [this](https://github.com/humphd/brackets/blob/bramble/src/editor/EditorManager.js#L225) function. When you hit CMD + E again to try and close the inline widget the call stack is the exact same as previously stated above. The [closeInlineWidget](https://github.com/humphd/brackets/blob/bramble/src/editor/EditorManager.js#L376) function is only called when you click on the "X" on the inline widget.

**Comment by AJDBenner** [comment]
I've been able to narrow this issue down further. The reason CMD + E isn't closing the inline widget is because the widget doesn't have focus. The focus is in the editor. 

If you click inside the widget then hit CMD + E it will close the quick edit. The desktop version of Brackets has this issue too. If you open the inline widget in desktop Brackets then click somewhere in the editor then try to hit CMD + E you can replicate this issue.

Solution: In desktop Brackets, once the inline widget is opened the focus is on the inline widget. In our version when the inline widget is opened the focus is still on the editor, not the widget. We need to give focus to the widget when CMD + E is pressed.

Here is what I mean

![demo](https://cloud.githubusercontent.com/assets/9154982/6855047/86c589e0-d3cb-11e4-9052-8f7c8414591e.gif)

You can see the cursor flashing in front of the "p" of the `<p>` tag the first time. After you click in the widget the cursor is no longer on the `<p>` tag and you see the "new rule" box highlight in blue.

**Comment by AJDBenner** [comment]
I've notice that in Bramble you aren't able to create a "new rule". I think this may be causing the issue. When you create a new CSS file then you have the option to create a "new rule" and this error doesn't occur. Here is what I mean.

![demo](https://cloud.githubusercontent.com/assets/9154982/6874844/c9a3958e-d490-11e4-8fae-4df2ee381b8b.gif)

**Comment by AJDBenner** [comment]
From what I've found the issue occurs [here](https://github.com/humphd/brackets/blob/bramble/src/editor/CSSInlineEditor.js#L317). The `disabled` class is added to the `New Rule` button until list is known. The issue is that the `disabled` class is never removed [here](https://github.com/humphd/brackets/blob/bramble/src/editor/CSSInlineEditor.js#L342). Unfortunately, this issue isn't fixed If you manually remove the `disabled` class.

**Comment by AJDBenner** [comment]
Fixed with https://github.com/humphd/brackets-browser-livedev/pull/70

**Comment by AJDBenner** [comment]
This bug has been filed upstream https://github.com/adobe/brackets/issues/10975
